pkg/image/graph/nodes/nodes.go

```diff
@@ -161,12 +161,27 @@ func FindOrCreateSyntheticImageStreamNode(g osgraph.MutableUniqueGraph, is *imag
 	).(*ImageStreamNode)
 }
 
-// EnsureImageLayerNode adds a graph node for the layer if it does not already exist.
-func EnsureImageLayerNode(g osgraph.MutableUniqueGraph, layer string) graph.Node {
-	return osgraph.EnsureUnique(g,
-		ImageLayerNodeName(layer),
+// EnsureImageComponentNode adds a graph node for the image config or layer if it does not already exist.
+func EnsureImageComponentNode(g osgraph.MutableUniqueGraph, name string, t ImageComponentType) graph.Node {
```

Comment:
The reason why I have merged both cases into single component is that theoretically, the same blob may be referred by an image as a config and by another image as a layer. The both cases need to be represented as a single node in the graph. Otherwise, they would be pruned separately - first as a config, second as a layer. The second prune would fail naturally.

We could keep them separated with different nodes but we'd have to make sure not to delete them twice.